lib/src/exceptions/logic/invalid_connection_exception.dart

```diff
@@ -0,0 +1,20 @@
+/// Copyright (C) 2023 Intel Corporation
+/// SPDX-License-Identifier: BSD-3-Clause
+///
+/// invalid_connection_exception.dart
+/// An exception that thrown when there is an invalid logic connection
+///
+/// 2023 May 20
+/// Author: Sanchit Kumar <sanchit.dabgotra@gmail.com>
+///
+
+import 'package:rohd/rohd.dart';
+import 'package:rohd/src/exceptions/rohd_exception.dart';
+
+/// An exception that thrown when a [Logic] is trying for an invalid connection.
+class InvalidConnectionException extends RohdException {
+  /// Creates an exception when a logic is trying for an invalid connect.
+  InvalidConnectionException(Logic a, Logic b)
+      : super('Signal "$a" trying to connect with signal "$b"'
```

Comment:
This error message is true, but might be confusing to someone unintentionally connecting a signal to itself.  Can we either make this message more informative/specific that it's connecting a signal to itself?  Or else add an additional message that can be provided with details for the reason?